_posts/2017-03-04-event-sourcing-is-not-a-choice.md

```diff
@@ -68,69 +68,71 @@ this is the possible set:
-does not use them often. But from time to time we have to show the information
-somewhere in GUI, or make a report based on the data stored as events. In
-this case, it is a problem.
+It's not a problem in case your application only stores events without using them too often. 
+But from time to time you have to show the information
+somewhere in a GUI, or you have to make a report based on the data stored as events. 
+In these cases you will encounter problems. 
 
-You don't have completed domain entity, you can only build it from events 
-when it is needed. The more changes you have the slower is the process. It will 
-be extremely slow if you need to operate with bunch of entities at the 
+If you haven't completed the domain entity, then you can only build it from events 
+when needed. The more changes you have, the slower the process becomes. It will 
+be extremely slow if you need to operate with a bunch of entities at the 
 same time. 
 
 Of course, you can apply some optimizations. For instance, ES works 
-well in [CQRS model](https://martinfowler.com/bliki/CQRS.html). In that case,
-you'll have write model and fast read model. That might solve performance issue
-but it can also introduce new issues. It is always a trade-off. 
+well in [CQRS model](https://martinfowler.com/bliki/CQRS.html). In this case,
+you'll have to write a model and a fast read model. That might solve performance issues,
+but on the flip side of the coin, it might also introduce new issues. 
+The procedure is ultimately always a trade-off. 
 
 **It is hard to test**
 
-Obviously, it is very hard to cover event-driven objects with unit tests, because
-they always require some context and the context requires connection to storage.
-Unit test becomes 'fat' and less readable.
+Obviously, it is very hard to cover event-driven objects with unit tests because
+they always require form some context - and context requires connection to storage.
+Unit test becomes 'overblown' and less readable.
 
 
 **What is it good for?**
 
 If you have preconditions like this:
-* It is critical for application to restore its state at any time
-* Writes dominate over reads
+
+* It is critical for the application to restore its state at any time
+* Writing is more important than reading
 * There is no responsible UI to show domain entities
 
-you can think about ES.
+Well, then perhaps you can think about ES.
 
 The best examples of applying event-driven architecture are 
 [Blockchain](https://en.wikipedia.org/wiki/Blockchain_(database)) and 
-[Bitcoin](https://en.wikipedia.org/wiki/Bitcoin). Of course, everything
-related to storing transactions is potential candidate. But not everything
-that moves.
+[Bitcoin](https://en.wikipedia.org/wiki/Bitcoin). Essentially, anything
+related to storing transactions could come to mind to here. However, as my blog post has shown, 
+ES becomes problematic when dealing with anything "on-the-move". 
```

Comment:
The old phrase, `But not everything that moves.`, was a bad attempt to translate Ukrainian idiom to English. I only wanted to say that event-driven architecture can't be applied everywhere. 

Can we change the last phrase to `But make sure not to apply event-driven architecture when it is not needed.`?

Reply:
Yeah, I was wondering about that one. I wasn't sure if it was Github jargon so I tried to do something along the lines of your original phrasing with movement... I changed it now to something that's more natural